test/likelihoods/negativebinomial.jl

```diff
@@ -1,16 +1,30 @@
 @testset "NegativeBinomialLikelihood" begin
-    for args in ((), (logistic,), (LogisticLink(),)), kwargs in ((), (; successes=1))
-        lik = NegativeBinomialLikelihood(args...; kwargs...)
-        @test lik isa NegativeBinomialLikelihood{LogisticLink,Int}
-        @test lik.successes == 1
+    rs = (10, 9.5)
+    # Test based on p success
+    for nbparam in (NBParamSuccess, NBParamFailure)
+        for r in (10, 9.5) # Test both input types
+            @testset "$(nameof(nbparam)), r=$r" begin
+                lik = NegativeBinomialLikelihood(nbparam(r))
+                @test lik isa NegativeBinomialLikelihood{<:nbparam,LogisticLink}
+                lik = NegativeBinomialLikelihood(nbparam(r), logistic)
+                @test lik isa NegativeBinomialLikelihood{<:nbparam,LogisticLink}
+                test_interface(lik, NegativeBinomial; functor_args=(:params, :invlink))
+            end
+        end
     end
-
-    for args in ((normcdf,), (NormalCDFLink(),)), kwargs in ((; successes=2.0),)
-        lik = NegativeBinomialLikelihood(args...; kwargs...)
-        @test lik isa NegativeBinomialLikelihood{NormalCDFLink,Float64}
-        @test lik.successes == 2
+    # Test based on mean = link(f)
+    for (nbparam, args) in
+        ((NBParamI, (2.0,)), (NBParamII, (3.0,)), (NBParamPower, (2.0, 2.0)))
```

Comment:
```suggestion
        ((NBParamI, (2.345,)), (NBParamII, (3.456,)), (NBParamPower, (2.34, 3.21)))
```
For semi-arbitrary parameters I much prefer to use numbers that aren't a simple integer